test/metabase/api/dashboard_test.clj

```diff
@@ -2026,6 +2027,83 @@
+                 {:field-name "date" :base-type :type/Date #_#_::good "2020-02-02"}
+                 {:field-name "datetime" :base-type :type/DateTime #_#_::good "2020-02-02 14:39:59"}
+                 ;; Difference between h2 and postgres, in and out
+                 {:field-name "datetimetz" :base-type :type/DateTimeWithTZ #_#_::good "2020-02-02 14:39:59-0700" ::bad "not date"}]]
+      (actions.test-util/with-temp-test-data
+        ["types"
+         (map #(dissoc % ::good ::bad) types)
+         [["init"]]]
+        (actions.test-util/with-actions-enabled
+          (mt/with-temp* [Card [{card-id :id} {:dataset true :dataset_query (mt/mbql-query types)}]
+                          ModelAction [_ {:slug "insert" :card_id card-id}]
+                          Dashboard [{dashboard-id :id}]
+                          DashboardCard [{dashcard-id :id} {:dashboard_id dashboard-id
+                                                            :card_id card-id
+                                                            :visualization_settings {:action_slug "insert"}}]]
+            (testing "Good data"
+              (doseq [{:keys [field-name] value ::good} (filter ::good types)]
+                (testing (str "Attempting to implicitly insert " field-name)
+                  (mt/user-http-request :crowberto :post 200  (format "dashboard/%s/dashcard/%s/execute/insert" dashboard-id dashcard-id)
+                                        {:parameters {field-name value}})
+                  (let [{{:keys [rows cols]} :data} (qp/process-query (mt/mbql-query types))]
+                    (is (partial= {field-name value}
+                                  (zipmap (map (comp str/lower-case :name) cols)
+                                          (last rows))))))
+                (actions.test-util/with-action [{action-id :action-id} (custom-action-for-field field-name)]
+                  (mt/with-temp* [ModelAction [_ {:slug "custom" :card_id card-id :action_id action-id}]
+                                  DashboardCard [{custom-dashcard-id :id} {:dashboard_id dashboard-id
+                                                                           :card_id card-id
+                                                                           :visualization_settings {:action_slug "custom"}}]]
+                    (testing (str "Attempting to custom insert " field-name)
+                      (mt/user-http-request :crowberto :post 200
+                                            (format "dashboard/%s/dashcard/%s/execute/custom" dashboard-id custom-dashcard-id)
+                                            {:parameters {field-name value}})
+                      (let [{{:keys [rows cols]} :data} (qp/process-query (mt/mbql-query types))]
+                        (is (partial= {field-name value}
+                                      (zipmap (map (comp str/lower-case :name) cols)
+                                              (last rows))))))))))
```

Comment:
I can't see how we know `value` is not coming from the implicit insert above. Does that value get removed somewhere?